### DISCUSSION TITLE
One knob function for modulations

### DISCUSSION THREAD

**Discussion by Banana71** [probonopd/MiniDexed#559] [Ideas]
Many musicians and sound artists want to influence the sound directly on the instrument. The Yamaha DX7 MKII has 2 continuous sliders CS1+CS2 that can be programmed to 105 different targets (see DX7II operating instructions page 39).
I would like a one knob function with which it would be possible to influence different parameters. e.g. cutoff, LFO speed or the existing modulation targets (Aftertouch, MW, Foot controller, Breath controller)

Operation in the miniDexed menu: Extension under the Modulation menu item:

 -  Modulation 
- - Allows to set (on and off) modulation wheel, foot controllers, breath controller and after touch targets. For any of this controllers there are three main components of voice to witch LFO modulation can be applied: to the PITCH, to the AMPLITUDE and to EG (Envelope Generator) BIAS. Each operator may or may not be programmed to be sensitive to modulation sources. Range parameter. Some settings could result in little or no audible sound, especially when EG BIAS is On.
-   - Mod. Wheel
- - - Range Allows to set the amount of effect that will be applied to one or more than one of Pitch, Amplitude, EG Bias. Can be set from 0 (no effect) to 99 (maximum effect)
- - - Pitch Can be set to Off or On (default)
- - - Amplitude Can be set to Off (default) or On
- - - EG Bias Envelope Generator Can be set to Off (default) or On
- - Foot Control (as Mod. Wheel above) but Pitch is Off (default)
- - Breath Control (as Mod. Wheel above) but Pitch is Off (default)
- - Aftertouch (as Mod. Wheel above) but Pitch is Off (default) 

- - One Knob
- - - Destination (off, cutoff, resonance, LFO-Speed, .....)
 Range min/max,  in the range the parameter is influenced (e.g. 60 - 99)

In addition, there should be a menu item in which the encoder takes over the One Knob function:
- Performance
- - Load
- - One Knob
- - Save
- - Delete

Or preferably in the new menu structure from @lexonic

In the example video, the sound of the miniDexed is influenced by aftertouch (brilliance) and at the same time I change the LFO speed via Dexed.

https://github.com/probonopd/MiniDexed/assets/104296205/8ff4017a-faf4-4beb-a356-51ce9334b0b4

[Performance: 000136_LFO-Spd2Saw.zip](https://github.com/probonopd/MiniDexed/files/13360876/000136_LFO-Spd2Saw.zip)

![Screenshot 2023-11-15 074252](https://github.com/probonopd/MiniDexed/assets/104296205/da29afb6-8ebe-4816-a93f-26934b06b3e0)

**Comment by Piattica** [comment]
all this could be done extending MIDI CC assignments

**Reply by diyelectromusic** [reply]
My dual MiniDexed setup includes the ability to send any UI control (apart from voice editing - I didn't add that one, although I still might) over MIDI using SysEx, so in theory that would allow anything in the UI to be controlled by an external device too.

Of course that would allow any replacement UI too just using MIDI...

Kevin
